crowbar_framework/app/models/service_object.rb

```diff
@@ -1034,10 +1045,10 @@ def apply_role(role, inst, in_queue, bootstrap = false)
       new_deployment.delete("elements_expanded")
     end
 
-    if skip_unready_nodes_enabled
+    unless cleaned_elements.nil?
       # if we have removed nodes from the list, make sure to expand them and overwrite the
       # new_elements var so we dont try to run chef-client on those not-ready nodes
-      new_elements, failures, msg = expand_items_in_elements(elements_without_unready)
+      new_elements, failures, msg = expand_items_in_elements(filtered_elements)
```

Comment:
`filtered_elements` is the old name of the variable ;-)

Reply:
fixed